src/PostgresLoggerDecorator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\LoggerTrait;
+
+final class PostgresLoggerDecorator implements LoggerInterface
+{
+    use LoggerTrait;
+
+    public const LOG_CATEGORY = 'db_app';
+    public const ORM_QUERY_LOG_CATEGORY = 'db_sys';
+
+    public function __construct(
+        private LoggerInterface $defaultLogger
+    ) {
+    }
+
+    public function log($level, string|\Stringable $message, array $context = []): void
+    {
+        $extendedContext = $this->extendContext($context, (string)$message);
+
+        $this->defaultLogger->log($level, $message, $extendedContext);
+    }
+
+    private function extendContext(array $context, string $message): array
+    {
+        if ($this->isPostgresSystemQuery($message)) {
+            $context['category'] = self::ORM_QUERY_LOG_CATEGORY;
+        } else {
+            $context['category'] = self::LOG_CATEGORY;
+        }
+
+        $context['time'] = empty($context['time']) ? microtime(true) : (float)$context['time'];
+
+        return $context;
+    }
+
+    protected function isPostgresSystemQuery(string $srcQuery): bool
+    {
+        $query = strtolower($srcQuery);
+
+        return
+            strpos($query, 'tc.constraint_name')
```

Comment:
Since PHP 8.0 is used, this could be replaced with `str_contains()`. With `strpos()` a strict check for `false` is needed (to handle `0` value).